_posts/2023-07-10-custom-http-signaling-notification.adoc

```diff
@@ -0,0 +1,249 @@
+
+Create a java project to build the `HTTP` signal and notification channels. Run the following command to create a new java project using Maven:
+
+[source,shell]
+----
+mvn archetype:generate
+    -DgroupId=io.debezium.examples
+    -DartifactId=http-signaling-notification
+----
+
+Add the following dependency to the `pom.xml` file with the Debezium version (2.3 and later versions):
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.debezium</groupId>
+    <artifactId>debezium-core</artifactId>
+    <version>${debezium.version}</version>
+</dependency>
+----
+
+To receive signals using a Mock Server, create a Docker Compose file that defines the Mock Server service. The configuration for the Mock Server service will be as follows:
+
+[source,yaml]
+----
+services:
+  mockServer:
+    image: mockserver/mockserver:latest
+    ports:
+      - 1080:1080
+    environment:
+      - MOCKSERVER_WATCH_INITIALIZATION_JSON=true
+      - MOCKSERVER_INITIALIZATION_JSON_PATH=/config/initializerJson.json
+    volumes:
+        - ./initializerJson.json:/config/initializerJson.json
+----
+
+The environment variables `MOCKSERVER_WATCH_INITIALIZATION_JSON` and `MOCKSERVER_INITIALIZATION_JSON_PATH` are set to enable the Mock Server to watch for changes in the initialization JSON file and specify its path. The `initializerJson.json` file, which contains the http request and response information for the signals, is mounted into the Mock Server container. The definition of the `initializerJson.json` file is as follows:
+
+[source,json]
+----
+[
+  {
+    "httpRequest" : {
+      "method" : "GET",
+      "path" : "/api/signal",
+      "queryStringParameters" : {
+        "code" : ["10969"]
+      }
+    },
+    "httpResponse" : {
+      "body": "{\"id\":\"924e3ff8-2245-43ca-ba77-2af9af02fa07\",\"type\":\"log\",\"data\":{\"message\": \"Signal message received from http endpoint.\"}}",
+      "statusCode": 200
+    }
+  }
+]
+----
+
+The `initializerJson.json` file defines a mock http request to the path `/api/signal` with the query string parameter `code=10969`. When this request is received by the Mock Server, it will respond with a JSON body containing an `id`, `type`, and `data`. The response has a status code of 200, indicating a successful response.
+
+Create the `HTTP` signal channel by extending the `SignalChannelReader` interface as shown below:
+
+[source,java]
+
+----
+public class HttpSignalChannel implements SignalChannelReader {
+
+        @Override
+    public String name() { // <1>
+        return "http";
+    }
+
+    @Override
+    public void init(CommonConnectorConfig connectorConfig) { // <2>
+        // custom configuration
+    }
+
+    @Override
+    public List<SignalRecord> read() { // <3>
+        // logic to read & process signal event from http endpoint
```

Comment:
Why not putting the real implementation?

Reply:
I wasn't sure to add it here or not, as the implementation is little longer.

Reply:
Ok.

Reply:
And we are providing this example's source code inside `debezium/examples`. But would also like to know @Naros's opinion on this.

Reply:
I concur with Mario, it's generally better from a follow-along style to include it, and given the full source isn't overly verbose I think it's fine.  Much like you've already done, you can always omit the imports to make the code more concise for the reader.